src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/MultiSourcePackageMetadataProvider.cs

```diff
@@ -52,27 +51,22 @@ public sealed class MultiSourcePackageMetadataProvider : IPackageMetadataProvide
             _logger = logger;
         }
 
-        public async Task<IPackageSearchMetadata> GetPackageMetadataAsync(PackageIdentity identity, bool includePrerelease, CancellationToken cancellationToken)
+        public async Task<IPackageSearchMetadata> GetPackageMetadataForIdentityAsync(PackageIdentity identity, CancellationToken cancellationToken)
```

Comment:
I see that [`MultiSourcePackageMetadataProvider`](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/MultiSourcePackageMetadataProvider.cs) implements [`IPackageMetadataProvider`](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/IPackageMetadataProvider.cs) interface.  

- The new public api `GetPackageMetadataForIdentityAsync` is not defined in the interface. Hence everytime we call this method `GetPackageMetadataForIdentityAsync`, we should check if the instance is of type `MultiSourcePackageMetadataProvider` before calling.

- The comment for existing `GetPackageMetadataAsync` method in the interface claims that `GetPackageMetadataAsync` method `Retrieves a package metadata of a specific version along with list of all available versions`
https://github.com/NuGet/NuGet.Client/blob/7f4d303c196a153be868d7ea57d12ba2581b926d/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/IPackageMetadataProvider.cs#L18-L26

Do we need to fix a bug in this original method implementation instead of creatring new api?

Reply:
Yes I was undecided if I wanted to add it to the interface or not, as that would be a breaking change. @zivkan  thoughts?

I don't believe there is a 'bug' there, but there is definitely some unintended consequences, the comment specifically says 'along with all available versions'.  This is the part that we are specifically trying to avoid, in the case of the 'test10k' package we end up spending an inordinate amount of time parsing JSON from over 150 200kb files (~15 seconds after some of my other changes, upwards of 50 seconds before my other changes) to get all the different versions available to return with this metadata.  In the PM UI we already have a list of all the other versions so this is pretty wasted.

Reply:
NuGet.PackageManagement.VisualStudio isn't part of the NuGet SDK, and therefore doesn't use the Public API Analyzers, so breaking changes aren't a concern (the solution will fail to compile if not resolved within our own repo).

Reply:
Thanks for the comment Andy, will modify the interface.